cci_nepal/analysis/dummy_data/dummy_data_modeling/model_analysis.py

```diff
@@ -0,0 +1,295 @@
+].tolist()
+yhat_basic = model_basic.predict([row_basic])
+
+# Summarizing prediction
+
+logging.info("Printing the scores for NFRI Basic")
+logging.info(yhat_basic[0])
+
+
+logging.info("Now printing the scores for NFRI Non Basic")
+
+
+# Making prediction
+
+row_non_basic = X.iloc[
+    10,
+].tolist()
+yhat_non_basic = model_non_basic.predict([row_non_basic])
+
+# Summarizing prediction
+
+logging.info(yhat_non_basic[0])
+
+
+# Plotting the Coefficient scores of the Model
+
+coefficient_scores = model_basic.coef_[0]
+
+for i, v in enumerate(coefficient_scores):
+    logging.info("Feature: %0d, Score: %.5f" % (i, v))
+
+
+logging.info(X.columns)
+pyplot.bar([x for x in range(len(coefficient_scores))], coefficient_scores)
+pyplot.show()
+
+coeff_basic_df = pd.DataFrame(model_basic.coef_[0], X.columns, columns=["Coefficient"])
+logging.info("The Coefficient for item", nfri_basic[0], "is:")
+coeff_basic_df
+
+# Getting the indices of items on the basis of importance in Descending order
+
+sorted_indices_basic = np.flip(np.argsort(yhat_basic[0]))
+
+sorted_indices_non_basic = np.flip(np.argsort(yhat_non_basic[0]))
+
+logging.info("NFRI Basic sorted on the basis of importance score")
+logging.info(itemgetter(*sorted_indices_basic)(nfri_basic))
+
+
+logging.info("Now time for NFRI Non Basic")
+logging.info(itemgetter(*sorted_indices_non_basic)(nfri_non_basic))
+
+logging.info("The importance label of above NFRI Basic")
+logging.info([label_transformer(d) for d in -np.sort(-yhat_basic[0])])
+
+logging.info("Now time for NFRI Non Basic")
+logging.info([label_transformer(d) for d in -np.sort(-yhat_non_basic[0])])
+
+# Evaluating multioutput regression model with k-fold cross-validation
+# Try one each for basic and non-basic respectively (in case of y that is)
+
+# Defining Model
+model_test = LinearRegression()
+# Setting the cross-validation parameters
+cv = RepeatedKFold(n_splits=10, n_repeats=3, random_state=1)
+# Evaluate the model and collecting the scores
+n_scores_basic = cross_val_score(
+    model_test, X, y_basic, scoring="neg_mean_absolute_error", cv=cv, n_jobs=-1
+)
+
+logging.info(n_scores_basic)
+# Making the scores positive
+n_scores_basic_absolute = absolute(n_scores_basic)
+# Summarizing the performance
+logging.info(
+    "MAE of Basic: %.3f (%.3f)" % (mean(n_scores_basic_absolute), std(n_scores_basic))
+)
+
+logging.info("Time for some cross validation score")
+
+r2_score_basic = cross_val_score(model_test, X, y_basic, cv=cv, scoring="r2")
+logging.info(r2_score_basic)
+logging.info("Average R2 Score")
+logging.info(np.nanmean(r2_score_basic))
+
+# Repeating the same steps for NFRI_non_basic
+
+n_scores_non_basic = cross_val_score(
+    model_test, X, y_non_basic, scoring="neg_mean_absolute_error", cv=cv, n_jobs=-1
+)
+logging.info(n_scores_non_basic)
+n_scores_non_basic_absolute = absolute(n_scores_non_basic)
+logging.info(
+    "MAE of Non Basic: %.3f (%.3f)"
+    % (mean(n_scores_non_basic_absolute), std(n_scores_non_basic))
+)
+
```

Comment:
This cell took a couple of minutes for me to run. Might just be my slow laptop but maybe worth highlighting or showing an output timer so the user is aware.